dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -164,25 +164,27 @@ def _check_dimension_associations(
         self, missing_dimension_associations: DataFrame | None
     ) -> None:
         """Check that a cross-join of dimension records is present, unless explicitly excepted."""
-        context = ScratchDirContext(Path(tempfile.gettempdir()))
-        assoc_by_records = self._make_expected_dimension_association_table_from_records(
-            [x for x in DimensionType if x != DimensionType.TIME], context
-        )
-        assoc_by_data = self._make_actual_dimension_association_table_from_data()
-        if missing_dimension_associations is None:
-            required_assoc = assoc_by_records
-        else:
-            required_assoc = filter_out_expected_missing_associations(
-                assoc_by_records, missing_dimension_associations
+        dsgrid_config = DsgridRuntimeConfig.load()
+        scratch_dir = dsgrid_config.scratch_dir or Path(tempfile.gettempdir())
+        with ScratchDirContext(scratch_dir) as context:
```

Comment:
Existing bug that I saw on a multi-node spark cluster. We need to ensure that all temp files are accessible to all nodes.